### PR TITLE
Feature/set port range

### PR DESCRIPTION
**Pull request by lbb**
Resolves #331

### CODE REVIEW

**Review by mrIncompetent — DISMISSED** [review]

File: cmd/kubermatic-api/cmd/root.go [hunk 1]
```diff
@@ -36,13 +38,31 @@ import (
...
 
 var viperWhiteList = []string{
 	"v",
 }
 
+func parseNodeRange() (minNodePort, maxNodePort int, err error) {
+	input := viper.GetString("node-port-range")
+	minMax := strings.Split(input, "-")
+	if len(minMax) != 2 {
+		return 1200, 14767, fmt.Errorf("can't parse %s, plsease enter it in the format of \"12000-14767\"", input)
```
Comment (scruplelesswizard) [inline cmd/kubermatic-api/cmd/root.go]: typo

File: cmd/kubermatic-api/cmd/root.go [hunk 2]
```diff
@@ -36,13 +38,31 @@ import (
...
+	}
+
+	min, err := strconv.Atoi(minMax[0])
+	if err != nil {
+		return 1200, 14767, err
```
Comment (scruplelesswizard) [inline cmd/kubermatic-api/cmd/root.go]: typo

File: cmd/kubermatic-api/cmd/root.go [hunk 3]
```diff
@@ -36,13 +38,31 @@ import (
...
+	}
+	max, err := strconv.Atoi(minMax[1])
+	if err != nil {
+		return 1200, 14767, err
```
Comment (scruplelesswizard) [inline cmd/kubermatic-api/cmd/root.go]: typo

File: cmd/kubermatic-api/cmd/root.go [hunk 4]
```diff
@@ -117,6 +143,7 @@ func init() {
 	RootCmd.PersistentFlags().StringVar(&jwtKey, "jwt-key", "", "The JSON Web Token validation key, encoded in base64")
 	RootCmd.PersistentFlags().StringVar(&address, "address", ":8080", "The address to listen on")
 	RootCmd.PersistentFlags().StringVar(&masterKubeconfig, "master-kubeconfig", "", "When set it will overwrite the usage of the InClusterConfig")
+	RootCmd.PersistentFlags().StringVar(&nodeRange, "node-port-range", "12000-14767", "The range node port will be allocated in, provider in format: \"12000-14767\"")
```
Comment (scruplelesswizard) [inline cmd/kubermatic-api/cmd/root.go]: Must explain what the port is used for. 
Also, this feature is not for a node port, this is the Client API server port. Please change the documentation and variable naming to reflect such

File: cmd/kubermatic-api/cmd/root.go [hunk 5]
```diff
@@ -36,13 +38,31 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-var cfgFile, kubeConfig, dcFile, secretsFile, jwtKey, address, masterKubeconfig string
+var cfgFile, kubeConfig, dcFile, secretsFile, jwtKey, address, masterKubeconfig, nodeRange string
```
Comment (scruplelesswizard) [inline cmd/kubermatic-api/cmd/root.go]: The variable naming is incorrect. Firstly, this sets the API Server Port Range, and has nothing at all to do with a node. Also, it does not reflect that it references ports at all...

**Review by scruplelesswizard — CHANGES_REQUESTED** [review]

File: cmd/kubermatic-api/cmd/root.go [hunk 6]
```diff
@@ -51,6 +71,12 @@ var RootCmd = &cobra.Command{
 	// Uncomment the following line if your bare application
 	// has an action associated with it:
 	Run: func(cmd *cobra.Command, args []string) {
+		minPort, maxPort, err := parseAPIServerPortRange()
+		if err != nil {
+			log.Fatalf("Error reading node range: %v", err)
```
Comment (scruplelesswizard) [inline cmd/kubermatic-api/cmd/root.go]: Node Range?

**Review by scruplelesswizard — CHANGES_REQUESTED** [review]

**Review by scruplelesswizard — APPROVED** [review]

**Pull request merged**